app/javascript/mastodon/features/ui/components/media_modal.js

```diff
@@ -84,14 +84,17 @@ export default class MediaModal extends ImmutablePureComponent {
       return null;
     }).toArray();
 
+    const containerStyle = {
+      alignItems: 'center', // center vertically
+    };
+
     return (
       <div className='modal-root__modal media-modal'>
         {leftNav}
 
         <div className='media-modal__content'>
           <IconButton className='media-modal__close' title={intl.formatMessage(messages.close)} icon='times' onClick={onClose} size={16} />
-
-          <ReactSwipeableViews onChangeIndex={this.handleSwipe} index={index} animateHeight>
+          <ReactSwipeableViews containerStyle={containerStyle} onChangeIndex={this.handleSwipe} index={index}>
```

Comment:
This is the major design change in this PR. `animateHeight` is disabled because `height` cannot be animated cheaply. It may be kinda-high-framerate on powerful desktop machines, but on mobile devices it looks pretty bad.

Because the height is not changing, I add `align-items: center` to the flexbox container so that the items are vertically centered. The design tradeoff is that there is now a letterboxing on the top and bottom of some images. I actually think this is an improvement, though, because it looks better when you're tapping-and-dragging to swipe left and right (IMO).